homeassistant/components/harmony/config_flow.py

```diff
@@ -0,0 +1,187 @@
+"""Config flow for Logitech Harmony Hub integration."""
+import logging
+from urllib.parse import urlparse
+
+import aioharmony.exceptions as harmony_exceptions
+from aioharmony.harmonyapi import HarmonyAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.components import ssdp
+from homeassistant.components.remote import (
+    ATTR_ACTIVITY,
+    ATTR_DELAY_SECS,
+    DEFAULT_DELAY_SECS,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_NAME): str}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    harmony = HarmonyAPI(ip_address=data[CONF_HOST])
+
+    _LOGGER.debug("harmony:%s", harmony)
+
+    try:
+        if not await harmony.connect():
+            await harmony.close()
+            raise CannotConnect
+    except harmony_exceptions.TimeOut:
+        raise CannotConnect
+
+    # As a last resort we get the name from the harmony client
+    # in the event a name was not provided
+    if CONF_NAME not in data or data[CONF_NAME] is None or data[CONF_NAME] == "":
+        data[CONF_NAME] = harmony.name
+
+    return {CONF_NAME: data[CONF_NAME], CONF_HOST: data[CONF_HOST]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Logitech Harmony Hub."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Harmony config flow."""
+        self.harmony_config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_HOST])
```

Comment:
This integration also supports discovery, using host is not a good way to specify a unique id. Don't harmony provide some kind of unique id?

Reply:
Good call. Yes they do. I fixed that in this PR so I'll break it out so it can be reused :)